docs/csharp/tour-of-csharp/tips-for-javascript-developers.md

```diff
@@ -0,0 +1,42 @@
+1. [***Pattern matching***](../fundamentals/functional/pattern-matching.md): Pattern matching enables concise conditional statements and expressions based on the shape of complex data structures. The [`is` statement](../language-reference/operators/is.md) checks if a variable "is" some pattern. The pattern-based [`switch` expression](../language-reference/operators/switch-expression.md) provides a rich syntax to inspect a variable and make decisions based on its characteristics.
+1. [***String interpolation***](../language-reference/tokens/interpolated.md) and [***raw string literals***](../language-reference/builtin-types/reference-types.md#string-literals): String interpolation enables you to insert evaluated expressions in a string, rather than using positional identifiers. Raw string literals provide a way to minimize escape sequences in text.
+1. [***Nullable and non-nullable types***](../nullable-references.md): C# supports *nullable value types*, and *nullable reference types* by appending the `?` suffix to a type. For nullable types, the compiler warns you if you haven't checked for `null` before dereferencing the expression. For non-nullable types, the compiler warns you if you might be assigning a `null` value to that variable. These features can minimize your application throwing a <xref:System.NullReferenceException?displayProperty=nameWithType>. The syntax may be familiar from TypeScript's use of `?` for optional properties.
+1. [***Properties***](../properties.md) and [***Indexers***](../indexers.md): Both properties and indexers (treating a class like an array or dictionary) have language support. In Java, they are naming conventions for methods starting with `get` and `set`.
+1. [***Extension methods***](../programming-guide/classes-and-structs/extension-methods.md):  In C#, you can create methods that *extend* a class or interface. These enable you to extend the behavior of a type from a library, or all types that implement a given interface.
+1. [***LINQ***](../linq/index.md): Language integrated query (LINQ) provides a common syntax to query and transform data, regardless of its storage.
+
+As you learn more, you'll find other differences as well, but many of those are smaller in scope.
+
+You will find familiar features and idioms from JavaScript and TypeScript that aren't available in C#:
+
+1. ***dynamic types***: C# uses static typing. A variable has a type when it is declared, and that type can't change.
+1. ***Prototypal inheritance***: C# inheritance is part of the type declaration. A C# `class` declaration states any base class. In JavaScript, you can set the `__proto__` property to set the base type on any instance.
+1. ***Interpreted language***: C# code must be compiled before it's run. JavaScript and TypeScript code can be run directly in the browswer.
```

Comment:
```suggestion
1. ***Interpreted language***: C# code must be compiled before it's run. JavaScript and TypeScript code can be run directly in the browser.
```

Reply:
I'm not sure I've seen Type run directly in the browser. How is that done?

Reply:
> I'm not sure I've seen TypeScript run directly in the browser. How is that done?

You can load the TypeScript compiler into the browser, and compile it. It's how the [TypeScript playground](https://www.typescriptlang.org/play) does it. For this article, I removed the note on TypeScript, as that's not really a common scenario.